ts/Data/Stores/DataStore.ts

```diff
@@ -114,43 +185,87 @@ implements DataEventEmitter<TEventObject>, DataJSON.Class {
         return json;
     }
 
+    /**
+     * The default load method, which fires the `afterLoad` event
+     * @emits DataStore#afterLoad
+     */
     public load(): void {
         fireEvent(this, 'afterLoad', { table: this.table });
     }
 
-    public whatIs(name: string): (DataStore.MetaColumn|void) {
-        const metadata = this.metadata;
-        let i;
-
-        for (i = 0; i < metadata.length; i++) {
-            if (metadata[i].name === name) {
-                return metadata[i];
-            }
-        }
+    /**
+     * Emits an event on the modifier to all registered callbacks of this event.
+     *
+     * @param {DataStore.EventObject} [e]
+     * Event object containing additional event information.
+     */
+    public emit(e: TEventObject): void {
+        fireEvent(this, e.type, e);
     }
 
+    /**
+     * Registers a callback for a specific modifier event.
+     *
+     * @param {string} type
+     * Event type as a string.
+     *
+     * @param {DataEventEmitter.EventCallback} callback
+     * Function to register for an modifier callback.
+     *
+     * @return {Function}
+     * Function to unregister callback from the modifier event.
+     */
     public on(
         type: TEventObject['type'],
         callback: DataEventEmitter.EventCallback<this, TEventObject>
     ): Function {
         return addEvent(this, type, callback);
     }
 
+    /**
+     * Converts the class instance to ClassJSON
+     */
     public abstract toJSON(): DataJSON.ClassJSON;
 
 }
 
+/* *
+ *
+ *  Namespace
+ *
+ * */
+
 namespace DataStore {
 
+    /**
+     * The JSON schema for datastore metadata
+     */
     export type MetadataJSON = Array<DataJSON.Array>;
 
+    /**
+     * The default event object for a datastore
+     */
     export interface EventObject extends DataEventEmitter.EventObject {
         readonly table: DataTable;
     }
 
-    export interface MetaColumn {
+    /**
+     * Metadata entry containing the name of the column
+     * and a metadata object
+     */
+    export interface MetaColumn extends DataJSON.Object {
         name?: string;
-        metadata?: any;
+        metadata?: Partial<MetadataType>;
+    }
+
+    /**
+     * A metadata object
+     */
+    export type MetadataType = {
+        title: string;
+        dataType: DataJSON.Types;
+        // validator: Function;
```

Comment:
Maybe a topic for the next meeting: how should we handle this and other functions in options?

Reply:
Yes, I agree. We should discuss this and also start with some guidelines in the readme to solve these kind of problems.